sgkit/tests/test_distance.py

```diff
@@ -64,36 +77,44 @@ def create_distance_matrix(
 
 
 @pytest.mark.parametrize(
-    "size, chunk",
+    "size, chunk, target",
     [
-        ((100, 100), (20, 10)),
-        ((100, 100), (25, 10)),
-        ((100, 100), (50, 10)),
+        ((100, 100), (20, 10), "cpu"),
```

Comment:
Given the specificity of the map/reduce funcs to chunkwise operations, I'd feel a little better about a test grid like this here:

```
((30, 30), (10, 10), "cpu"),
((30, 30), (10,  5), "cpu"),
((30, 30), ( 5, 10), "cpu")),
((15, 30), (10, 10), "cpu"),
((15, 30), (10,  5), "cpu"),
((15, 30), ( 5, 10), "cpu")),
((30, 15), (10, 10), "cpu"),
((30, 15), (10,  5), "cpu"),
((30, 15), ( 5, 10), "cpu")),
```

And the same for a gpu device.

I'm assuming that will still be quite fast.

Reply:
What happened with this one?  I don't see a resolution for it.

Reply:
I added this in here: https://github.com/pystatgen/sgkit/pull/498/files#diff-bfebf015a7d0a911d66743edcf9806a15b4661814667c0683fef0af8a6222b35R104

Probably a mistake, you requested in the correlation test and I added in the euclidean test, I think there is value in adding it in both, I'll add.

Reply:
I see, can you please collapse both of them into something like this?

```python
@pytest.mark.parametrize(
    "size, chunk, device, metric",
    [
        pytest.param(shape, chunks, device, metric, marks=pytest.mark.gpu if device == "gpu" else None)
        for shape in [(30, 30), (15, 30), (30, 15)]
        for chunks in [(10, 10), (5, 10), (10, 5)]
        for device in ["cpu", "gpu"]
        for metric in ["euclidean", "correlation"]
    ]
)
def test_pairwise(size, chunk, device, metric):
    ... # create `x`
    actual = pairwise_distance(x, metric=metric, device=device)
    expected = pdist(x, metric=metric)
    np.testing.assert_almost_equal(actual, expected)
```

Reply:
Makes sense, done thanks!